proposals/undersampling_project_proposal.md

```diff
@@ -0,0 +1,47 @@
+#### SIG TFX-Addons
+
+# Project Proposal
+
+------
+
+**Your name:** Daniel Kim
+
+**Your email:** danielk@twitter.com
+
+**Your company/organization:** Twitter
+
+**Project name:** Undersampling Component
+
+## Project Description
+
+This project will be a custom function-based component that inputs an artifact in `tfRecord` format of `tf.Example`s and randomly undersamples it, reducing the data to the lowest-frequency class. It will primarily use an underlying Apache Beam pipeline that will be wrapped inside the TensorFlow component.
```

Comment:
(non blocking)

How much do you imagine this implementation will be tied to either the format or the tf.Example payload? If it becomes a different container, or a different payload, would sample still work?

Reply:
While this implementation is tied to the tf.Example payload, this is the standard pipeline data flow for pipelines that are downstream from ExampleGen. We are planning for data format logic to be handled by ExampleGen; as long as your ExampleGen component or a custom component can ingest a requested data format, the Undersampler can use it.